main.tf

```
@@ -71,6 +71,37 @@ resource "aws_dynamodb_table" "this" {
     kms_key_arn = var.server_side_encryption_kms_key_arn
   }
 
+  dynamic "import_table" {
+    for_each = length(var.import_table) > 0 ? [var.import_table] : []
+
+    content {
+      input_format           = import_table.value.input_format
+      input_compression_type = try(import_table.value.input_compression_type, null)
+
+      dynamic "input_format_options" {
+        for_each = import_table.value.input_format == "CSV" ? try([import_table.value.input_format_options], []) : []
```

```suggestion
        for_each = try([import_table.value.input_format_options], [])
```

Let's not make assumptions about `CSV` but instead render the block if user specify the value of `input_format_options`.

This will be forward-compatible with future improvements if AWS/HashiCorp decides to do so.